### ISSUE TITLE
Hook or event when local database is created

### ISSUE THREAD

**Issue by bmihelac** [hoodiehq/discussion#116]
Local database is created on first page load, on sign in, and on sign out. It would be nice to have hook or event that would allow custom code to be performed when local database is created. 

For example database indexes could be created at that time or some data could be bootstrapped.

Currently database is cleared in after 'signout' hook (of hoodie-client/init).

```javascript
 hoodie.account.hook.after('signout', function (options) {
   return hoodie.store.reset()
 })
```

This make impossible to use `clear` event or `signout` hook to bootstrap new database. I have made an non working [signout hook example](https://github.com/bmihelac/hoodie-app-tracker/commit/e658624b909b8645ed63656d66a875b8cef5e97c) on hoodie-app-tracker app.

Currently to achieve running code is only possible when resolving`hoodie.account.signOut` as demonstrated in this commit: https://github.com/bmihelac/hoodie-app-tracker/commit/d0974b31d9664840cf26aa022b90aaf5ef9270ca

What do you think about emitting event 'store.created` or adding hook `hoodie.store.hook.after('database:create')`?

**Comment by janl** [comment]
Great idea!

I can’t comment on whether to use an event or a hook, but either sounds fine to me.

**Comment by gr2m** [comment]
I wonder if there are other use cases than indexes for which we would want the hook. If it would only be for indexes, I could imagine a declarative way. Indeces would be relevant for both, client and server

**Comment by bmihelac** [comment]
Maybe bootstraping some data?

**Comment by gr2m** [comment]
I’m currently thinking about a `bootstrap` event and hook. `bootstrap` would be emitted each time the database is "ready". On page reload, this will be nearly immediate. After sign in, we will emit the event after all user data has been synchronised to the client. This would also address the problem there is currently no way to tell when this data bootstrapping is done.

So to add custom data or views, you could then do

```js
hoodie.account.hook.before('bootstrap', function (options) {
   return hoodie.store.add(myAppViews)
 })
```

Would that work for you?

**Comment by bmihelac** [comment]
`bootstrap` would be emitted after db is recreated after destroyed, ie. when user signs out?

**Comment by gr2m** [comment]
Yes, that’s my thinking. Here the list of cases

1. Page load: emits once database is ready
2. Sign in: emits once all data is loaded from remote
3. Sign out: emits once new database is ready

**Comment by bmihelac** [comment]
This looks good. Thanks
